### PR TITLE
feat: [EXT-2970] space api deprecation warnings

### PR DESCRIPTION
**Pull request by Jwhiles**
# Purpose of PR
image
image
image

## Todo
- [x] ensure warnings show up in VSCODE etc

## PR Checklist

- [ ] Tests are added/updated/not required
- [ ] Tests are passing
- [ ] Typescript typings are added/updated/not required

### CODE REVIEW

**Review by colomolo — APPROVED** [review]
LGTM

**Review by shikaan — APPROVED** [review]

File: lib/types/space.types.ts [hunk 1]
```diff
@@ -32,86 +32,291 @@ type Snapshot<T> = {
...
+   * @deprecated since version 4.0.0
+   * Consider using TODO - what method?
+   * See https://www.contentful.com/developers/docs/extensibility/app-framework/sdk/#using-the-contentful-management-library for more details
+   */
   waitUntilAssetProcessed: (assetId: string, locale: string) => Promise<Asset>
 
-  /** Returns all users who belong to the space. */
+  /**
+   * @deprecated since version 4.0.0
+   * Consider using `cma.user.getManyForSpace` or `cma.user.getManyForOrganization` instead
```
Comment (kdamball) [inline lib/types/space.types.ts:245]: can you do `getManyForOrg`? considering the space env scope of apps

**Review by kdamball — APPROVED** [review]
LGTM but left a few questions mostly with the release schedule of this feature :)

File: lib/types/space.types.ts [hunk 2]
```diff
@@ -32,86 +32,291 @@ type Snapshot<T> = {
...
+   * @deprecated since version 4.0.0
+   * Consider using `cma.scheduledActions.getMany` instead
+   * See https://www.contentful.com/developers/docs/extensibility/app-framework/sdk/#using-the-contentful-management-library for more details
+   * Returns a list of scheduled actions for the currenst space & environment
+   */
   getAllScheduledActions: () => Promise<ScheduledAction[]>
 
+  /**
+   * @deprecated since version 4.0.0
+   * Consider using TODO - which method?
```
[the remark this answers is not in the document]
Comment (Jwhiles) [reply lib/types/space.types.ts]: This method just polled the get endpoint before, so I've just suggested the same. WDYT?

File: lib/types/space.types.ts (hunk 1 above)
Comment (Jwhiles) [reply lib/types/space.types.ts:245]: no! good point. :)

**Pull request merged**